src/jsonspec/pointer/stages.py

```diff
@@ -54,10 +54,10 @@ def stage(obj, parent=None, member=None):
     """
     obj = Staged(obj, parent, member)
 
-    if isinstance(obj, dict):
+    if isinstance(obj, MutableMapping):
         for key, value in obj.items():
             stage(value, obj, key)
-    elif isinstance(obj, (list, tuple)):
+    elif isinstance(obj, MutableSequence):
```

Comment:
``MutableSequence``works for list and they declinaisons, but not for ``tuple``! Maybe you should test with ``Sequence`` and exclude string types (you can use six for these)